### DISCUSSION TITLE
Not all sliders working, like close eyes - I assume controlled by blink?

### DISCUSSION THREAD

**Discussion by alankent** [pixiv/three-vrm#1550] [Help]
I have sliders going for all blend shapes in a face editor, but some don't work. For example, Fcl_EYE_Joy_R works

![image](https://github.com/user-attachments/assets/adeccccf-5370-42f2-9b57-2164ab867101)

But Fcl_EYE_CLose_R does not.

![image](https://github.com/user-attachments/assets/23d62a9b-fbb6-4031-a79e-e36bcfb05efb)

I assume this is because of a blink behavior taking over those controls (?). But its also the Fcl_ALL_* blend shapes. I thought these worked before, so its a bit confusing.

![image](https://github.com/user-attachments/assets/1e01cfca-0a92-46f5-97ea-b4a2fa7cc72c)

Does setting Fcl_EYE_Fun weight to zero override the Fcl_ALL_Fun somehow? If so, are there programatic rules I can use to decide which weights to set to zero? Currently I default all weights to zero.

Thank you for any advice!

**Comment by 0b5vr** [comment]
Please provide a minimal reproducible demo. We can't determine that it's a problem of the avatar, the app, or three-vrm.

**Comment by alankent** [comment]
I have a suspicion of what it is. I added vrm.update() in a lot more places, so I think it is the expression manager overriding the blend shape weights for any blend shape used by an expression. I think I need to work out how to turn off the expression manager while allowing the user to define new expressions by adjusting blend shape weights - or create a new expression and have the sliders update that expression definition, and see if edits to the expression definition are sent through immediately to blend shape weights.

Thanks for the quick response. I have some more things to try.
